lib/rspec/core/formatters/helpers.rb

```diff
@@ -70,8 +70,7 @@ def self.format_seconds(float, precision=nil)
         # @param string [String] string with trailing zeros
         # @return [String] string with trailing zeros removed
         def self.strip_trailing_zeroes(string)
-          stripped = string.sub(/[^1-9]+$/, '')
-          stripped.empty? ? "0" : stripped
+          string.sub(/(?:(\..*[^0])0+|\.0+)$/, '\1')
```

Comment:
No problem, added link to Rubular, with few more specific examples. Basically looks for a decimal point followed by any number of non-zeros, then a zero (or more). Or just a decimal point and zeros. Then replaces this with the first match (i.e. the non-zeros)